### ISSUE TITLE
ProximityZone with custom range below 0.1 not detected.

### ISSUE THREAD

**Issue by idybil** [Estimote/Android-Proximity-SDK#47]
## Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * [x] My beacons have Estimote Monitoring enabled (Check it in the Estimote Cloud, or via app)
    * [X] My Android device/devices supports BLE and has Android OS version >= 5.0.0 
    * [X] My Android device/devices have bluetooth enabled
    * [X] My app has Location Permissions granted
    
## Basic information

**Estimote SDK version:** [0.5.0]

**Android devices affected:** [Samsung Galaxy S4]

**Android OS version affected:** [Lollipop 5.0.1]

**Beacon hardware version:** [G1.8] 

## Description

** Steps to reproduce:**

When custom range is 0.1, the beacon is detected. Change the custom range to 0.05 or 0.09, then the beacon is no more detected.

**Expected behavior:** [Beacon should be detected when custom range is below 0.1]

**Actual behavior:** [Beacon is not detected when custom range is below 0.1]

## Additional information
```
We need to detect beacon when the phone is nearly touching the beacon, that is why we need a custom range below 0.1.   Custom range below 0.1 is working well using the iOS Proximity SDK version 0.12.0
```

**Comment by pawelDylag** [comment]
Hey @idybil 

Sorry for the delay, I was on my vacation last week. 

As for your case - the BLE technology is based on electro-magnetic waves with frequencies that are really prone to interference. You should not expect any interaction based on this technology to have super hight precision. This is why we are switching the mindset to use `far range` and `near range` rather than exact distance values (which will never be super exact due to the nature of this tech).

tl;dr:
1. If your use case needs the phone to be nearly touching a beacon, consider using NFC technology rather than BLE. 
2. The method `withCustomRange` has a rounding floor operator applied to avoid using super-precise values. When you set `0.1` the value gets rounded to `0.1`, but when you set `0.05` it gets rounded to `0.0`, resulting in the absence of detection. 

We do not consider this as an issue, but rather as an internal feature. Anyway it should be better documented and I will add a better explanation to this method. 

Cheers, 
Paweł

**Comment by idybil** [comment]
Thanks for your answer @pawelDylag . I understand your mindset. I just wondering if you do not round the custom range (0.05 to 0.0) will the beacon will be detected ? if yes, why do the rounding ? since it won't affect the far - near anyway.

Please let me know if you will still "wontfix" this issue.

**Comment by pawelDylag** [comment]
@idybil as I said - `a rounding floor operator is applied to avoid using super-precise values`.  The minimal defined precision is 0.1m, so putting values that does not mach this precision will result in rounding to apply. All values <`0.1` are an edge case, that is resolved to just simply `0.0`.

I think you should really consider using [NFC in our beacons](https://developer.estimote.com/proximity/nfc/), because even if you put `0.000001 m` range, it may still be triggered on 1 m on some phones due to differences in antennas, surrounding, etc. 

Cheers, 
Paweł

**Comment by idybil** [comment]
OK, thanks for your support @pawelDylag , I will have a look at NFC. You can close this issue.
